src/Catalyst.Core.Modules.Dfs/CoreApi/UnixFsApi.cs

```diff
@@ -0,0 +1,465 @@
+                header.Size = 0;
+                tar.PutNextEntry(entry);
+                tar.CloseEntry();
+            }
+            else // Must be a file
+            {
+                var content = await ReadFileAsync(cid, cancel).ConfigureAwait(false);
+                header.TypeFlag = TarHeader.LF_NORMAL;
+                header.Size = content.Length;
+                tar.PutNextEntry(entry);
+                await content.CopyToAsync(tar, cancel);
+                tar.CloseEntry();
+            }
+
+            // Recurse over files and subdirectories
+            if (dm.Type == DataType.Directory)
+            {
+                foreach (var link in dag.Links)
+                {
+                    await AddTarNodeAsync(link.Id, $"{name}/{link.Name}", tar, cancel).ConfigureAwait(false);
+                }
+            }
+        }
+
+        IBlockApi GetBlockService(AddFileOptions options)
+        {
+            return options.OnlyHash
+                ? new HashOnlyBlockService()
+                : _blockApi;
+        }
+
+        /// <summary>
+        ///   A Block service that only computes the block's hash.
+        /// </summary>
+        class HashOnlyBlockService : IBlockApi
+        {
+            public IPinApi PinApi { get; set; }
+
+            public Task<IDataBlock> GetAsync(Cid id, CancellationToken cancel = default)
+            {
+                throw new NotImplementedException();
+            }
+
+            public Task<Cid> PutAsync(byte[] data,
+                string contentType = Cid.DefaultContentType,
+                string multiHash = MultiHash.DefaultAlgorithmName,
+                string encoding = MultiBase.DefaultAlgorithmName,
+                bool pin = false,
+                CancellationToken cancel = default)
+            {
+                var cid = new Cid
+                {
+                    ContentType = contentType,
+                    Encoding = encoding,
+                    Hash = MultiHash.ComputeHash(data, multiHash),
+                    Version = (contentType == "dag-pb" && multiHash == "sha2-256") ? 0 : 1
+                };
+                return Task.FromResult(cid);
+            }
+
+            public Task<Cid> PutAsync(Stream data,
+                string contentType = Cid.DefaultContentType,
+                string multiHash = MultiHash.DefaultAlgorithmName,
+                string encoding = MultiBase.DefaultAlgorithmName,
+                bool pin = false,
+                CancellationToken cancel = default)
+            {
+                throw new NotImplementedException();
+            }
+
+            public Task<Cid> RemoveAsync(Cid id,
+                bool ignoreNonexistent = false,
+                CancellationToken cancel = default)
+            {
+                throw new NotImplementedException();
+            }
+
+            public Task<IDataBlock> StatAsync(Cid id, CancellationToken cancel = default)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused private method 'StatAsync'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)